include/circt/Dialect/FIRRTL/FIRRTLTypes.h

```diff
@@ -59,6 +59,9 @@ class FIRRTLType : public Type {
   /// types recursively within itself.
   bool isPassive() { return getRecursiveTypeProperties().isPassive; }
 
+  /// Returns true if this is a "passive" that which is not analog.
+  bool isNonAnalogPassive() { return isPassive() && !isa<AnalogType>(); }
```

Comment:
I think this might not quite work as intended when the type is an analog inside a bundle, i.e.
`!firrtl.bundle<a : analog<1>>`.  Luckily we already have `containsAnalog()` a couple lines below or it could be defined as:
```c++
bool isNonAnalogPassive() {
  auto props = getRecursiveTypeProperties();
  return !props. isPassive && !props.containsAnalog;
}
```
We should add a test to `name-preservation.fir` for it: `wire w: {a: Analog<1>}`

Reply:
The FIRRTL spec doesn't recursively ban analog unfortunately and there's a CIRCT test asserting it is allowed in bundles.